R/class-CorrelationResult.R

```diff
@@ -0,0 +1,34 @@
+check_correlation_result <- function(object) {
+    errors <- character()
+
+    if (!length(object@data1Metadata) > 0) {
+      msg <- "Compute result must include information about data1 in the data1Metadata slot"
+      errors <- c(errors, msg)
+    } else if (!length(object@data2Metadata) > 0) {
+      msg <- "Compute result must include information about data2 in the data2Metadata slot"
+      errors <- c(errors, msg) 
+    }
```

Comment:
i guess were okay w an empty data.frame here for the stats? also maybe we can call the statistics slot something more specific, since we know the type of stat were computing here?

Reply:
I thought about this a bit. I decided to keep it "statistics" for consistency with the other compute results and because once we add p values and adj p values it will also contain statistics. 
Diff abund also has its statistics slot return a value plus some pvalue and adjpvals. Maybe both of these slots should be called 'data' instead of statistics? Something to think about in relation to #58 as well